sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,143 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        storage_config: Dict[Literal["size", "storage_class"], str] = None,
+        model_provider_parameters: HuggingFaceModelParams = None,
+        dataset_provider_parameters: Union[HfDatasetParams, S3DatasetParams] = None,
+        train_parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Dict[Literal["gpu", "cpu", "memory"], any] = None,
+    ):
+        """
+        Higher level train api
+        """
+        if (
+            not name
+            or not storage_config
+            or not model_provider_parameters
+            or not dataset_provider_parameters
+            or not train_parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        namespace = namespace or self.namespace
+
+        if "cpu" not in resources_per_worker or "memory" not in resources_per_worker:
+            raise ValueError("cpu and memory resources not specified")
+        else:
+            limits = {
+                "cpu": resources_per_worker["cpu"],
+                "memory": resources_per_worker["memory"],
+            }
+
+        if (
+            resources_per_worker["gpu"] is not None
+            and num_procs_per_worker > resources_per_worker["gpu"]
+        ) or (resources_per_worker["gpu"] is None and num_procs_per_worker != 0):
+            raise ValueError("Insufficient gpu resources allocated to the container.")
+
+        if "gpu" in resources_per_worker:
+            limits["nvidia.com/gpu"] = resources_per_worker["gpu"]
+
+        requests = limits.copy()
+
+        try:
+            self.core_api.create_namespaced_persistent_volume_claim(
+                namespace=namespace,
+                body=utils.get_pvc_spec(
+                    pvc_name=constants.TRAINER_PVC_NAME,
+                    namespace=namespace,
+                    storage_size=storage_config["size"],
+                    storage_class=storage_config["storage_class"],
+                ),
+            )
+        except Exception as e:
+            print(e)
```

Comment:
Please can you rise an Exception, similar to this one: https://github.com/kubeflow/training-operator/blob/7802b2dc3391c40dadd7e675d778f1bef03285ab/sdk/python/kubeflow/training/api/training_client.py#L421